src/VisualStudio/LiveShare/Impl/RunCodeActionsHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    /// <summary>
+    /// Run code actions handler.  Called when lightbulb invoked.
+    /// TODO - Move to CodeAnalysis.LanguageServer once the UI thread dependency is removed.
+    /// </summary>
+    [ExportLspRequestHandler(LiveShareConstants.RoslynContractName, LSP.Methods.WorkspaceExecuteCommandName)]
+    internal class RunCodeActionsHandler : CodeActionsHandlerBase, ILspRequestHandler<LSP.ExecuteCommandParams, object, Solution>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        public RunCodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringService codeRefactoringService, IThreadingContext threadingContext)
+            : base(codeFixService, codeRefactoringService)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<object> HandleAsync(LSP.ExecuteCommandParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+        {
+            var solution = requestContext.Context;
+            if (request.Command.StartsWith(RemoteCommandNamePrefix))
```

Comment:
do you know what these mean?

Reply:
I've asked, pending response, for now I've filed a work item https://github.com/dotnet/roslyn/projects/45#card-20873029